### ISSUE TITLE
Notifications double processed

### ISSUE THREAD

**Issue by BottlecapDave** [BottlecapDave/Bottlecap.Net.Notifications#2]
When calling ScheduleAndExecuteAsync, the notification is being sent, but the record is not being marked as successful/failed. This results in the background task picking it up and processing again.

**Comment by BottlecapDave** [comment]
EntityFramework implementation needs to be updated to be

```
var addedNotifications = new List<INotificationData>();
            foreach (var notification in data)
            {
                var entity = await _context.Notifications.AddAsync(notification);
                addedNotifications.Add(entity.Entity);
            }

            var numberOfChanged = await _context.SaveChangesAsync();
            return numberOfChanged > 0 ? addedNotifications : null;
```

**Comment by BottlecapDave** [comment]
This should be fixed in v1.0.1
